src/compiler/scanner.ts

```diff
@@ -1127,7 +1116,7 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
      * `pos` is outside the bounds set for `text`, `CharacterCodes.EOF` is returned instead.
      */
     function codePointChecked(pos: number) {
-        return pos >= 0 && pos < end ? codePointUnchecked(pos) : CharacterCodes.EOF;
+        return pos < end ? codePointUnchecked(pos) : CharacterCodes.EOF;

```

Comment:
I carefully examined all expressions involving `pos--` and `pos -= number` and determined that the `pos >= 0` check is unnecessary.